lib/Interpreter/Transaction.cpp

```diff
@@ -68,7 +68,12 @@ namespace cling {
           if (name.equals(ND->getNameAsString()))
             return ND;
         }
-        else if (LinkageSpecDecl* LSD = dyn_cast<LinkageSpecDecl>(DI)) {
+      }
+    }
+    // Not found yet, peek inside extern "C" decalrations
+    for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
```

Comment:
@vgvassilev Even after the first ADL aided find, the function is *still* not really in the global namespace:
```
root [1] f(a)
I got called!!!
(int) 0
root [2] f(3)
input_line_15:2:3: error: use of undeclared identifier 'f'
 (f(3))
  ^
```